examples/ui/overflow_debug.rs

```diff
@@ -0,0 +1,303 @@
+            image: UiImage::new(asset_server.load("branding/bevy_logo_dark_big.png")),
+            style: Style {
+                size: Size::new(Val::Auto, Val::Px(100.)),
+                position_type: PositionType::Absolute,
+                top: Val::Px(-50.),
+                left: Val::Px(-200.),
+                ..default()
+            },
+            ..default()
+        });
+    });
+}
+
+fn spawn_text(
+    parent: &mut ChildBuilder,
+    asset_server: &Res<AssetServer>,
+    update_transform: impl UpdateTransform + Component,
+) {
+    spawn_container(parent, update_transform, |parent| {
+        parent.spawn(TextBundle::from_section(
+            "Bevy",
+            TextStyle {
+                font: asset_server.load("fonts/FiraSans-Bold.ttf"),
+                font_size: 120.0,
+                color: Color::WHITE,
+            },
+        ));
+    });
+}
+
+fn spawn_container(
+    parent: &mut ChildBuilder,
+    update_transform: impl UpdateTransform + Component,
+    spawn_children: impl FnOnce(&mut ChildBuilder),
+) {
+    let mut transform = Transform::default();
+
+    update_transform.update(0.0, &mut transform);
+
+    parent
+        .spawn((
+            NodeBundle {
+                style: Style {
+                    size: Size::new(Val::Px(CONTAINER_SIZE), Val::Px(CONTAINER_SIZE)),
+                    align_items: AlignItems::Center,
+                    justify_content: JustifyContent::Center,
+                    overflow: Overflow::Hidden,
+                    ..default()
+                },
+                background_color: Color::DARK_GRAY.into(),
+                ..default()
+            },
+            Container(0),
+        ))
+        .with_children(|parent| {
+            parent
+                .spawn((
+                    NodeBundle {
+                        style: Style {
+                            align_items: AlignItems::Center,
+                            justify_content: JustifyContent::Center,
+                            top: Val::Px(transform.translation.x),
+                            left: Val::Px(transform.translation.y),
+                            ..default()
+                        },
+                        transform,
+                        ..default()
+                    },
+                    update_transform,
+                ))
+                .with_children(spawn_children);
+        });
+}
+
+fn update_animation(
+    mut animation: ResMut<AnimationState>,
+    time: Res<Time>,
+    keys: Res<Input<KeyCode>>,
+) {
+    let time = time.elapsed_seconds();
+
+    if keys.just_pressed(KeyCode::Space) {
+        animation.playing = !animation.playing;
+
+        if !animation.playing {
+            animation.paused_at = time;
+        } else {
+            animation.paused_total += time - animation.paused_at;
+        }
+    }
+
+    if animation.playing {
+        animation.t = (time - animation.paused_total) % LOOP_LENGTH / LOOP_LENGTH;
+    }
+}
+
```

Comment:
Yup, I know `transform` should be derived from style… but didn't find any other way to change scale/rotation. I wanted to force scale/rotation (translation it's used to update top/left) to see how it behaves.

Although, now that I think about it… maybe it doesn't make any sense to put these in the example if it's not a current feature. Or… it could be used to fix scale/rotation? Although, probably this should be done at shader/stencil/mask/whatever… level; using textures and composition and not by cutting geometry. (?)

Reply:
I've been working on #8240 which fixes some of the problems, but I'm not too sure about the API though. 
It's not necessarily a problem that the example cases don't all display correctly, we could just put a WIP label or something. Users need to be aware that the clipping is very basic.

Reply:
Hey, I had a surgery a couple of days ago, I don't think I'll check the PRs anytime soon. Please fork it and do as you wish.

Reply:
I think its reasonable to merge this as-is. Just make whatever changes are needed in #8240.